pkg/adapterManager/manager.go

```diff
@@ -121,9 +100,68 @@ func newManager(r builderFinder, m map[aspect.Kind]aspect.Manager, exp expr.Eval
 	}
 }
 
-// Execute iterates over cfgs and performs the actions described by the combined config using the attribute bag on each config.
-func (m *Manager) Execute(ctx context.Context, cfgs []*configpb.Combined,
-	requestBag *attribute.MutableBag, responseBag *attribute.MutableBag, ma aspect.APIMethodArgs, df descriptor.Finder) aspect.Output {
+// Check dispatches to the set of aspects associated with the Check API method
+func (m *Manager) Check(ctx context.Context, requestBag *attribute.MutableBag, responseBag *attribute.MutableBag) rpc.Status {
+	return m.dispatch(ctx, requestBag, responseBag, checkMethod,
+		func(wrapper aspect.Wrapper, evaluator expr.Evaluator) rpc.Status {
+			cw := wrapper.(aspect.CheckWrapper)
+			return cw.Execute(requestBag, evaluator)
+		})
+}
+
+// Report dispatches to the set of aspects associated with the Report API method
+func (m *Manager) Report(ctx context.Context, requestBag *attribute.MutableBag, responseBag *attribute.MutableBag) rpc.Status {
+	return m.dispatch(ctx, requestBag, responseBag, reportMethod,
+		func(wrapper aspect.Wrapper, evaluator expr.Evaluator) rpc.Status {
+			rw := wrapper.(aspect.CheckWrapper)
+			return rw.Execute(requestBag, evaluator)
+		})
+}
+
+// Quota dispatches to the set of aspects associated with the Quota API method
+func (m *Manager) Quota(ctx context.Context, requestBag *attribute.MutableBag, responseBag *attribute.MutableBag,
+	qma *aspect.QuotaMethodArgs) (rpc.Status, *aspect.QuotaMethodResp) {
+
+	var qmr *aspect.QuotaMethodResp
+	o := m.dispatch(ctx, requestBag, responseBag, quotaMethod,
+		func(wrapper aspect.Wrapper, evaluator expr.Evaluator) rpc.Status {
+			qw := wrapper.(aspect.QuotaWrapper)
+			var o rpc.Status
+			o, qmr = qw.Execute(requestBag, evaluator, qma)
+			return o
+		})
+
+	return o, qmr
+}
+
+type invokeWrapperFunc func(wrapper aspect.Wrapper, evaluator expr.Evaluator) rpc.Status
+
+// Execute resolves config and invokes the specific set of aspects necessary to service the current request
+func (m *Manager) dispatch(ctx context.Context, requestBag *attribute.MutableBag, responseBag *attribute.MutableBag,
+	method apiMethod, invokeFunc invokeWrapperFunc) rpc.Status {
+	// get a new context with the attribute bag attached
+	ctx = attribute.NewContext(ctx, requestBag)
+
+	cfg, _ := m.cfg.Load().(config.Resolver)
+	if cfg == nil {
+		// config has not been loaded yet
+		const msg = "Configuration is not yet available"
+		glog.Error(msg)
+		return status.WithInternal(msg)
+	}
+
+	cfgs, err := cfg.Resolve(requestBag, m.methodMap[method])
+	if err != nil {
+		msg := fmt.Sprintf("unable to resolve config: %v", err)
```

Comment:
I'm deferring this for now (this is just copy/pasted code after all). This is tracked by issue #393